components/create_post/create_post.jsx

```diff
@@ -1113,7 +1113,10 @@ export default class CreatePost extends React.Component {
         if (readOnlyChannel) {
             createMessage = Utils.localizeMessage('create_post.read_only', 'This channel is read-only. Only members with permission can post here.');
         } else {
-            createMessage = Utils.localizeMessage('create_post.write', 'Write a message...');
+            createMessage = formatMessage(
+                {id: 'create_post.write', defaultMessage: 'Write to {channelDisplayName}'},
+                {channelDisplayName: currentChannel.display_name}
```

Comment:
How about for DM/GM channels?  Should it display according to TeammateNameDisplay setting?

Reply:
I tested with DMs, but not sure I tested with GMs. DMs should show display name of the team member.

Reply:
Oh right, the `channel.display_name` is a pre-processed field on redux that considers teammate name display setting.